packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -105,8 +112,16 @@ void main() {
 }
 ''');
 
+    int observatoryPort;
```

Comment:
We do have code in `os.dart` - `findAvailablePort()` - that tries to find an available local port. I'm not sure it's more reliable than the random generator below however -